pkg/sdk/reconciler/reconciler.go

```diff
@@ -387,14 +387,7 @@ func (r *Reconciler) CheckForOrphans(logger logr.Logger, cr client.Object) (bool
 func (r *Reconciler) CrUpdate(phase sdkapi.Phase, cr client.Object) error {
 	status := r.crManager.Status(cr)
 	status.Phase = phase
-	statusCopy := new(sdkapi.Status)
-	status.DeepCopyInto(statusCopy)
-	err := r.client.Update(context.TODO(), cr)
-	if err != nil {
-		return err
-	}
-	statusCopy.DeepCopyInto(status)
-	return r.client.Status().Update(context.TODO(), cr)
+	return r.client.Update(context.TODO(), cr)
```

Comment:
Kind of confused by the original implementation of this which calls `Update()` and `Status().Update()`.  I assume this is to cover the case when the status subresource is implemented on the CRD?  If so, I would suggest a `crUpdateStatus` method instead.  Nevertheless, this is the issue I encountered:

Looks like the k8s fake client code was updated recently.  Now, if update is called and the resource has a deletionTimestamp and no finalizers, the resource is deleted.  That wasn't the case before.  Only Delete() would delete the resource.  This was causing an issue in cdi unit tests when Reconcile() was passed a resource with deletiontimestamp set.  The call to `Update` here was deleting the resource so the call to `Status().Update()` was returning not found.

I feel that since the this library adds/removes finalizers the controller needs `Update` permission.  If the controller wants to support the status subresouce that should be an opt in thing that is configured

Reply:
I think the reason for two calls is that `Update` does not update the status sub-resource of the CR and there needs to be separate `Status().Update()` call to cover these changes.

Reply:
Okay, but there are a couple issues here:

1.  For CRDs that don't have subresource enabled (like cdi) what happens when `status.Update()` is called?  Is that a no-op request to the k8s apiserver?  If so, we should attempt to get rid of that inefficiency
2. For unit tests, it is an error to call `status.Update()` if deletionTimestamp != nil and len(finalizers) == 0

2 can easily be addressed in a hacky way, but what about 1?  Should the api of the library be updated to support subresource in an efficient way?  Like add `SubresourceEnabled() bool` to crManager interface?

Reply:
@jakub-dzon I refactored to better accomidate status. subresource support.  What do you think?